Selenium Readme (#169)
apify/actor-templates: templates/python-selenium/README.md

> # Python - Selenium & Chrome Actor template

Reviewer: I don't think it needs to have the `Python` prefix here, we don't have that pattern for other templates. In CLI you need to choose the language so you already know it's Python templates and on template detail we have it the category in the sidebar.

```suggestion
# Selenium & Chrome Actor template
```

## Included features

- **[Apify SDK](https://docs.apify.com/sdk/python/)** - toolkit for building Apify Actors
- **[Input schema](https://docs.apify.com/platform/actors/development/input-schema)** - define and easily validate a schema for your Actor's input
- **[Request queue](https://docs.apify.com/sdk/python/docs/concepts/storages#working-with-request-queues)** - queues into which you can put the URLs you want to scrape
- **[Dataset](https://docs.apify.com/sdk/python/docs/concepts/storages#working-with-datasets)** - store structured data where each object stored has the same attributes

## How it works

> This code is a Python script that uses the Selenium to scrape web pages and extract data from them. Here's a brief overview of how it works:

Reviewer 2: ```suggestion
This code is a Python script that uses Selenium to scrape web pages and extract data from them. Here's a brief overview of how it works:
```

> This code is a Python script that uses the Apify platform to scrape web pages and extract data from them. Here's a brief overview of how it works:

Reviewer 2: This is not really true, the script doesn't use the Apify platform if you run it locally. I would replace "Apify platform" with "Selenium" here.

# Python - Selenium & Chrome Actor template

> A template example built with Selenium and headless Chrome browser to scrape a website and save the results to storage. The URL of the web page is passed in via input, which is defined by the [input schema](https://docs.apify.com/platform/actors/development/input-schema). The template uses the [Selenium WebDriver](https://www.selenium.dev/documentation/webdriver/) to get the HTML of the page. Enqueued URLs are available in [request queue](https://docs.apify.com/sdk/python/reference/class/RequestQueue). The data are then stored in a [dataset](https://docs.apify.com/platform/storage/dataset) where you can easily access them.

Reviewer 2: ```suggestion
A template example built with Selenium and headless Chrome browser to scrape a website and save the results to storage. The URL of the web page is passed in via input, which is defined by the [input schema](https://docs.apify.com/platform/actors/development/input-schema). The template uses the [Selenium WebDriver](https://www.selenium.dev/documentation/webdriver/) to load and process the page. Enqueued URLs are stored in the default [request queue](https://docs.apify.com/sdk/python/reference/class/RequestQueue). The data are then stored in the default [dataset](https://docs.apify.com/platform/storage/dataset) where you can easily access them.
```

## Included features

> - **[Apify SDK](https://docs.apify.com/sdk/python/)** - a toolkit for building actors

Reviewer: ```suggestion
- **[Apify SDK](https://docs.apify.com/sdk/python/)** - toolkit for building Apify Actors

> - **[Input schema](https://docs.apify.com/platform/actors/development/input-schema)** - define and easily validate a schema for your actor's input

Reviewer 2: ```suggestion
- **[Input schema](https://docs.apify.com/platform/actors/development/input-schema)** - define and easily validate a schema for your Actor's input
```

- **[Request queue](https://docs.apify.com/sdk/python/docs/concepts/storages#working-with-request-queues)** - queues into which you can put the URLs you want to scrape

> - **[Dataset](https://docs.apify.com/sdk/python/docs/concepts/storages#working-with-datasets)** - store structured data where each object stored has the same attributes

Reviewer: How about adding the Selenium here too? It's the main feature of this template

Author: I thought this was generally related to Apify features, at least that's how it works in other templates. Also, it's mentioned in paragraph above.

## How it works
This code is a Python script that uses the Apify platform to scrape web pages and extract data from them. Here's a brief overview of how it works:

> - The script reads the input data from the Actor instance, which is expected to contain a start_urls key with a list of URLs to scrape and a max_depth key with the maximum depth of nested links to follow.

Reviewer: Let's use the `code` formatting for the variable names everywhere in the description (`start_urls`, `max_depth`,..) - it makes reading easier and users would know you're referencing to something in the code

- The script enqueues the starting URLs in the default request queue and sets their depth to 1.
- The script processes the requests in the queue one by one, fetching the URL using requests and parsing it using Selenium.
- If the depth of the current request is less than the maximum depth, the script looks for nested links in the page and enqueues their targets in the request queue with an incremented depth.

> - The script extracts the desired data from the page (in this case, titles of each page) and pushes it to the default dataset using the push_data method of the Actor instance.

Reviewer 2: ```suggestion
- The script extracts the desired data from the page (in this case, titles of each page) and pushes them to the default dataset using the push_data method of the Actor instance.
```

- The script catches any exceptions that occur during the scraping process and logs an error message using the Actor.log.exception method.

> - This code demonstrates how to use Python and the Apify SDK to scrape web pages and extract specific data from them.

Reviewer 2: This is kind of redundant, same as line 13
